controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -51,6 +53,10 @@ import (
 	"sigs.k8s.io/cluster-api/util/secret"
 )
 
+type RemoteClusterClientGetter interface {
```

Comment:
xref https://github.com/kubernetes-sigs/cluster-api/blob/72a7b2f1d9df0170ad5a43580ca2991b384f6e8b/bootstrap/kubeadm/controllers/kubeadmconfig_controller.go#L69-L70 for how this is handled in the bootstrap controller

Reply:
Updated